org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/correction/GetterSetterQuickFixTest.java

```diff
@@ -0,0 +1,423 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.correction;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author qisun
+ *
+ */
+public class GetterSetterQuickFixTest extends AbstractQuickFixTest {
+	private IJavaProject fJProject;
+	private IPackageFragmentRoot fSourceFolder;
+
+	@Before
+	public void setup() throws Exception {
+		fJProject = newEmptyProject();
+		fJProject.setOptions(TestOptions.getDefaultOptions());
+		fSourceFolder = fJProject.getPackageFragmentRoot(fJProject.getProject().getFolder("src"));
+	}
+
+	@Test
+	public void testInvisibleFieldToGetterSetter() throws Exception {
+		IPackageFragment pack = fSourceFolder.createPackageFragment("test", false, null);
+		StringBuilder buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    private byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		ICompilationUnit cu = pack.createCompilationUnit("C.java", buf.toString(), false, null);
+
+		buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		Expected e1 = new Expected("Change visibility of 'test' to 'package'", buf.toString());
+
+		buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    private byte test;\n");
+		buf.append("\n");
+		buf.append("    /**\n");
+		buf.append("     * @return the test\n");
+		buf.append("     */\n");
+		buf.append("    public byte getTest() {\n");
+		buf.append("        return test;\n");
+		buf.append("    }\n");
+		buf.append("\n");
+		buf.append("    /**\n");
+		buf.append("     * @param test the test to set\n");
+		buf.append("     */\n");
+		buf.append("    public void setTest(byte test) {\n");
+		buf.append("        this.test = test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
```

Comment:
no, that should be `c.setTest(c.getTest()+1)`

Reply:
The first quick fix suggestion will be ```Create setter and getter for test```, after that, ```Replace c.test with setter``` will show up to fix the problem. The behavior is consistent with eclipse.  Are we going to show all the quick fix at one time?

![fix_invisible](https://user-images.githubusercontent.com/5179063/38965884-d4cd790a-43b1-11e8-8ec2-875f7df10bb1.gif)

Reply:
The default behavior I see in Eclipse is to use those accessors directly.

![create-accessors](https://user-images.githubusercontent.com/148698/38966789-f5d20644-4351-11e8-8987-6c2b4bc65266.gif)

It makes no sense to force users to have a 2-step fix when it could be done at once

Reply:
BTW the 2nd code action you showed ("use setter") should still be available, precisely in case where the accessors already exist